### PR TITLE
Article search

### PR DESCRIPTION
**Pull request by angrysun**
added AJAX search function for articles by title, content, or author (can remove author or content if we don't want that)

### CODE REVIEW

**Comment by angrysun** [comment]
Screen Shot 2022-02-23 at 22 41 01

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -1,7 +1,18 @@
 class ArticlesController < ApplicationController
   def index
-    @articles = policy_scope(Article).all
+    @query = params[:query]
+    @articles = policy_scope(Article).order(created_at: :desc)
+    if params[:query].present?
+      @articles = Article.search_by_title_content_and_author(params[:query])
+    else
+      @article = Article.all
```
Comment (CamillieFu) [inline app/controllers/articles_controller.rb:8]: @articles?

File: app/models/article.rb [hunk 2]
```diff
@@ -1,4 +1,15 @@
 class Article < ApplicationRecord
   acts_as_taggable_on :tags
   validates :title, :date, :author, :content, :image, presence: true
+
+  include PgSearch::Model
+  pg_search_scope :search_by_title_content_and_author,
+                  against: {
+                    title: 'A',
```
Comment (CamillieFu) [inline app/models/article.rb:8]: maybe content could be first? and perhaps lets remove author as we won't have any unfortunately ...

**Review by CamillieFu — CHANGES_REQUESTED** [review]
Looks good! Just a few tiny things then it should be good to go!

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (angrysun) [reply app/controllers/articles_controller.rb:8]: Good catch! Will change that.

File: app/models/article.rb (hunk 2 above)
Comment (angrysun) [reply app/models/article.rb:8]: Sounds good, will change it up

**Comment by angrysun** [comment]
@CamillieFu Requested changes completed

**Pull request merged**